src/dom/Value.ts

```diff
@@ -309,6 +343,99 @@ export function Value<Clazz extends Constructor>(
+     *        equivalent to Z.
+     *     3. reflexive: X is equivalent to X.
+     *
+     * Ordered Sequence Equivalence:
+     * When an ordered sequence (i.e. tuple) of elements is specified in this
+     * document, equivalence over such an ordered sequence is defined as follows.
+     *
+     * A tuple, A = (a1, a2, ..., an), is equivalent to another tuple, B = (b1, b2,
+     * ..., bm) if and only if the cardinality (number of elements) of A equals the
+     * cardinality of B (i.e. n == m) and ai is equivalent to bi for i = 1 ... n.
+     *
+     * Un-Ordered Sequence Equivalence:
+     * When an un-ordered sequence (i.e. bag or multi-set) is specified in this
+     * document, equivalence over such a sequence is defined as follows.
+     *
+     * A bag, A = {a1, a2, ..., an} is equivalent to B = {b1, b2, ..., bm} if and
+     * only if the cardinality of A is equal to the cardinality of B and for each
+     * element, x, in A there exists a distinct element, y, in B for which x is
+     * equivalent to y.
+     *
+     * Values:
+     * Any arbitrary, atomic value in the Ion Language can be denoted as the tuple,
+     * (A, V), where A is an ordered list of annotations, and V is an Ion Primitive
+     * Data or Ion Complex Data value. The list of annotations, A is an tuple of Ion
+     * Symbols (a specific type of Ion Primitive).
+     */
+    // Class expressions (like this mixin) cannot have private or protected methods.
+    _ionEquals(
+      expectedValue: any,
+      options: {
+        epsilon?: number | null;
+        ignoreAnnotations?: boolean;
+        ignoreTimestampPrecision?: boolean;
+        onlyCompareIon?: boolean;
+      } = {
+        epsilon: null,
+        ignoreAnnotations: false,
+        ignoreTimestampPrecision: false,
+        onlyCompareIon: true,
+      }
+    ): boolean {
+      this._unsupportedOperation("ionEquals");
+    }
+
+    equals(
+      expectedValue: any,
+      options: {
+        epsilon?: number | null;
+        ignoreAnnotations?: boolean;
+        ignoreTimestampPrecision?: boolean;
+        onlyCompareIon?: boolean;
+      } = {
+        epsilon: null,
+        ignoreAnnotations: false,
+        ignoreTimestampPrecision: false,
+        onlyCompareIon: true,
+      }
+    ): boolean {
+      if (options.onlyCompareIon === false) {
+        options.ignoreAnnotations = true;
+        options.ignoreTimestampPrecision = true;
+      }
+      if (!options.ignoreAnnotations) {
+        if (!(expectedValue instanceof Value)) {
+          return false;
+        }
+        let actualAnnotations = this.getAnnotations();
+        let expectedAnnotations = (expectedValue as Value).getAnnotations();
+        if (actualAnnotations.length !== expectedAnnotations.length) {
+          return false;
+        }
+        for (let i = 0; i < actualAnnotations.length; i++) {
+          if (actualAnnotations[i].localeCompare(expectedAnnotations[i])) {
```

Comment:
* What's the motivation to use `localeCompare` here instead of `===`?
* While this logic is correct, I had to look up the return value description for `localeCompare` to understand why. I suggest adding a ` !== 0` at the end of this to make your expectations clear.